[CL-3784] Bugfix: Mentions endpoint no longer returns inactive users

jinjagit opened the pull request:
Please watch the short video in the [ticket description](https://citizenlab.atlassian.net/browse/CL-3784) for details of the bug this fixes.

# Changelog
## Fixed
- [CL-3784] The user mention mechanism, used when a user starts typing @some_user_name, for example in a comment, no longer returns inactive user(s) in the list used to select which user to mention. This could cause the app to become unresponsive.


[CL-3784]: https://citizenlab.atlassian.net/browse/CL-3784?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jinjagit committed:
[CL-3784] Bugfix & regression test

sebastienhoorens commented on back/spec/acceptance/mentions_spec.rb:
```diff
@@ -45,5 +45,15 @@
     end
+
+    example 'Does not return invited user by (partial) mention', document: false do
+      users.first.update(invite_status: 'pending')
```
Use update! instead of update to ensure that the update will truly happen.

sebastienhoorens commented on back/spec/acceptance/mentions_spec.rb:
```diff
@@ -45,5 +45,15 @@
+
+      do_request
+
+      expect(response_status).to eq 200
```
You could use assert_status 200 here :)

sebastienhoorens reviewed (changes requested):
Looks good to me, and makes sense. I even wonder if we could consider using active? instead (do we want to filter out other users, like blocked users perhaps?), but maybe that's not necessary.

Just one requested change: use `update!`.

jinjagit committed:
[CL-3784] Filter out inactive users in mentions endpoint + tests

jinjagit commented:
@sebastienhoorens Thanks!

I switched to using `.active`, as you suggested, changed the test to be a test of unregistered user + added a test of blocked user.

sebastienhoorens reviewed (approved):
👍
